File: app/src/main/java/com/innso/mobile/ui/viewModels/CompanyViewModel.java
package com.innso.mobile.ui.viewModels;


import android.view.View;

import com.innso.mobile.ui.activities.expenses.AddExpenseActivity;
import com.innso.mobile.ui.activities.NewCustomerActivity;
import com.innso.mobile.ui.activities.UsersActivity;
import com.innso.mobile.ui.activities.bills.BillsActivity;
import com.innso.mobile.ui.activities.expenses.ExpensesActivity;
import com.innso.mobile.viewModels.models.StartActivityModel;

public class CompanyViewModel extends BaseViewModel {


    public CompanyViewModel() {
        getComponent().inject(this);
    }

    public void addUser(View view) {
        startActivityEvent.onNext(new StartActivityModel(UsersActivity.class));
    }

    public void addCustomer(View view) {
        startActivityEvent.onNext(new StartActivityModel(NewCustomerActivity.class));
    }

    public void showBills(View view) {
        startActivityEvent.onNext(new StartActivityModel(BillsActivity.class));
    }

    public void addExpense(View view) {
        startActivityEvent.onNext(new StartActivityModel(ExpensesActivity.class));
    }

}

File: app/src/main/java/com/innso/mobile/viewModels/company/CompanyViewModel.kt
package com.innso.mobile.viewModels.company


import android.view.View
import com.innso.mobile.ui.activities.NewCustomerActivity
import com.innso.mobile.ui.activities.UsersActivity
import com.innso.mobile.ui.activities.bills.BillsActivity
import com.innso.mobile.ui.activities.expenses.ExpensesActivity
import com.innso.mobile.viewModels.AndroidViewModel
import com.innso.mobile.viewModels.models.StartActivityModel

class CompanyViewModel : AndroidViewModel() {

    fun addUser(view: View) =
            startActivity.postValue(StartActivityModel(UsersActivity::class.java))

    fun addCustomer(view: View) =
            startActivity.setValue(StartActivityModel(NewCustomerActivity::class.java))

    fun showBills(view: View) =
            startActivity.setValue(StartActivityModel(BillsActivity::class.java))

    fun addExpense(view: View) =
            startActivity.setValue(StartActivityModel(ExpensesActivity::class.java))
}
